### ISSUE TITLE
Error 500

### ISSUE THREAD

**Issue by balamut-andrey** [go-gitea/gitea#6475]
- Gitea version (or commit ref): 1.5.0-dev (?)
- Git version: git version 2.11.0 
- Operating system: Debian 9
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL (mysql  Ver 15.1 Distrib 10.1.26-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2)
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant
- Log gist:

gitea.log:
`2019/03/31 04:00:19 [...routers/user/auth.go:90 checkAutoLogin()] [E] AutoSignIn: GetUserByName: invalid connection`

## Description

New installation, 1 user, 0 repositories.
Appears in a variety of places. When you refresh the page, the error disappears.
I installed from source code.gitea,io, the domain is linked via nginx, there are no errors in nginx (error.log).

Help, please and sorry for my english.

**Comment by balamut-andrey** [comment]
`2019/03/31 04:18:01 [...itea/routers/home.go:149 RenderRepoSearch()] [E] SearchRepositoryByName: Count: invalid connection`

`2019/03/31 04:22:09 [...ules/context/repo.go:305 func1()] [E] GetUserByName: invalid connection`

**Comment by zeripath** [comment]
Gitea 1.5 is now very old. Please could you update to 1.7.5. Our development is very rapid and there have been a number of security issues discovered since 1.5.

**Comment by balamut-andrey** [comment]
Yesterday I executed the code `go get -d -u code.gitea.io/gitea` command. So, I have an old version? Or is my version not 1.5.0-dev?

At the bottom of the page is written 1.5.0-dev.
Or where better to download? From here?
Here, too, the file `https://github.com/go-gitea/gitea/blob/master/main.go` says (line 27):
 
`// Version holds the current Gitea version`
`var Version = "1.5.0-dev"`

I'm confused.

**Comment by zeripath** [comment]
So am I!

Ok how did you build Gitea? What `TAGS` did you set? The invalid connection is saying that there is something wrong with the database connection.

If you're using sqlite you need to build:

```bash
TAGS="bindata sqlite sqlite_unlock_notify" make generate build
```

**Comment by balamut-andrey** [comment]
I didn’t use any tags when installing, I just executed command `go build`. Maybe I need to specify some tags?

I do not use sqlite, I use MySQL .
phpMyAdmin works fine, does not lose connections.

**Comment by zeripath** [comment]
Ok please read: https://docs.gitea.io/en-us/install-from-source/

**Comment by balamut-andrey** [comment]
Yes, I watched it.
I do not need these tags.

**Comment by lunny** [comment]
It seems sometimes connecting your mysql databases failed?

**Comment by balamut-andrey** [comment]
In var/log/mysql/error.log there are no errors during the appearance of Error-500. The database is located on the same server as Gitea.

**Comment by balamut-andrey** [comment]
In the file var/log/syslog, I found some more such errors:
`gitea[366]: [Macaron] 2019-04-01 04:31:24: Started POST /user/login for 31.23.135.100`
`gitea[366]: [mysql] 2019/04/01 04:31:24 packets.go:36: unexpected EOF`
`gitea[366]: [Macaron] 2019-04-01 04:31:24: Completed POST /user/login 500 Internal Server Error in 5.099803ms`
`gitea[366]: [Macaron] 2019-04-01 04:31:24: Started GET /img/500.png for 31.23.135.100`
`gitea[366]: [Macaron] [Static] Serving /img/500.png`
`gitea[366]: [Macaron] 2019-04-01 04:31:24: Completed GET /img/500.png 304 Not Modified in 1.222161ms`

.../log/gitea.log
`2019/04/01 04:31:24 [...routers/user/auth.go:140 SignInPost()] [E] UserSignIn: invalid connection`

Maybe this concerns my problem? https://github.com/go-sql-driver/mysql/issues/674
What should I do?

**Comment by lunny** [comment]
I think gitea should add `SetConnMaxLifetime()` setting on xorm and if you run a recent version of gitea. `go-sql-driver/mysql#674` should have been resolved.

**Comment by zeripath** [comment]
I think this should have been resolved by #7051 by @lunny. Therefore I'm going to close this.
